Add endpoint context API (#1731)
withastro/docs: src/pages/en/reference/api-reference.md

</ul>
```

## Endpoint Context

[Endpoint functions](/en/core-concepts/endpoints/) receive a context object as the first parameter. It mirrors many of the `Astro` global properties.

```ts title="endpoint.json.ts"
import { APIContext } from 'astro';

export function get(context: APIContext) {
  // ...
}
```

### `context.params`

> `context.params` is an object passed from `getStaticPaths()` used to prerender [dynamic routes](/en/core-concepts/routing/#dynamic-routes).

Author: Yup this is more accurate!

```ts title="src/pages/posts/[id].json.ts"
import { APIContext } from 'astro';

export function getStaticPaths() {
  return [
    { params: { id: '1' } },
    { params: { id: '2' } },
    { params: { id: '3' } }
  ];
}

export function get({ params }: APIContext) {
	return {
		body: JSON.stringify({ id: params.id })
	};
}
```

See also: [`params`](#params)

> 

Author: Ah I forgot about `props`. Thanks!

The `getStaticPaths()` function should return an array of objects to determine which paths will be pre-rendered by Astro.

> It can also be used in [static file endpoints](/en/core-concepts/endpoints/#static-file-endpoints) and [server endpoints](/en/core-concepts/endpoints/#server-endpoints-api-routes) in `server` mode.

Reviewer: Right, you can't use it in SSR - my [PR here](https://github.com/withastro/docs/pull/1709) tries to make that more clear!

> It can also be used in [static file endpoints](/en/core-concepts/endpoints/#static-file-endpoints) and [server endpoints](/en/core-concepts/endpoints/#server-endpoints-api-routes) in `server` mode.

Author: Yeah I don't think `getStaticPaths` make sense in SSR since we're not doing anything static. If I've implied that, my mind may have went mush when writing the docs 😬

> It can also be used in [static file endpoints](/en/core-concepts/endpoints/#static-file-endpoints) and [server endpoints](/en/core-concepts/endpoints/#server-endpoints-api-routes) in `server` mode.

Reviewer 2: Ah! Right, that matches more with what I expected. So maybe we adjust this to something like this:

```suggestion
It can be used in [static file endpoints](/en/core-concepts/endpoints/#static-file-endpoints) as well as in `.astro` pages.
```

The `getStaticPaths()` function should return an array of objects to determine which paths will be pre-rendered by Astro.

> It can also be used in static file endpoints for [dynamic routing](d/en/core-concepts/endpoints/#params-and-dynamic-routing).

Reviewer 2: Oops, not sure what happened here, but this should fix it:

```suggestion
It can also be used in static file endpoints for [dynamic routing](/en/core-concepts/endpoints/#params-and-dynamic-routing).
```
